includes/Modules/AdSense.php

```diff
@@ -842,4 +801,19 @@ protected function setup_services( Google_Client $client ) {
 			'adsense' => new Google_Service_AdSense( $client ),
 		);
 	}
+
+	/**
+	 * Gets the module's Settings instance.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Settings AdSense\Settings instance.
+	 */
+	public function get_settings() {
```

Comment:
I opted for the trait, but rather than using a setter, it has an abstract protected `get_settings_instance` method for creating the instance which is memoized by the `get_settings` method. I think this is cleaner than messing with the constructor and still allows for the benefit of initializing the class when needed even though the module settings are always registered at the same time.